File: feature-main/src/main/java/com/smedialink/feature_main/di/MainActivityModule.java
package com.smedialink.feature_main.di;

import android.support.annotation.NonNull;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

import com.smedialink.feature_main.view.MainActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class MainActivityModule {

    @Provides
    @NonNull
    public FragmentManager provideFragmentManager(MainActivity mainActivity) {
        return mainActivity.getSupportFragmentManager();
    }

    @Provides
    @NonNull
    public AppCompatActivity provideAppCompatActivity(MainActivity mainActivity) {
        return mainActivity;
    }


}

File: feature-main/src/main/java/com/smedialink/feature_main/di/MainActivityModule.kt
package com.smedialink.feature_main.di

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import com.smedialink.feature_main.view.MainActivity
import dagger.Module
import dagger.Provides

@Module
class MainActivityModule {

    @Provides
    fun provideFragmentManager(mainActivity: MainActivity): FragmentManager {
        return mainActivity.supportFragmentManager
    }

    @Provides
    fun provideAppCompatActivity(mainActivity: MainActivity): AppCompatActivity {
        return mainActivity
    }


}
